src/jxl.imageio/jxlinput.cpp

```diff
@@ -0,0 +1,366 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// JPEG XL
+
+// https://jpeg.org/jpegxl/index.html
+// https://jpegxl.info
+// https://jpegxl.info/test-page
+// https://people.csail.mit.edu/ericchan/hdr/hdr-jxl.php
+// https://saklistudio.com/jxltests
+// https://thorium.rocks
+// https://bugs.chromium.org/p/chromium/issues/detail?id=1451807
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <jxl/decode.h>
+#include <jxl/decode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (0)
+
+class JxlInput final : public ImageInput {
+public:
+    JxlInput() { init(); }
+    ~JxlInput() override { close(); }
+    const char* format_name(void) const override { return "jpeg xl"; }
```

Comment:
No space, please. People might use the format_name to construct filenames or other uses where spaces will not be good. Let's call it "jpegxl".

Reply:
Understood. I mentioned non breaking space instead. However I am OK with jpegxl name. Will modify ASAP.